benchmark/README.md

```diff
@@ -0,0 +1,56 @@
+
+
+## install.py
+`python install.py` should handle
+- install dependencies
+- download & preprocess dataset (if applicable)
+
+Note:
+- `torch` should not be a dependency, it has to work with latest pytorch master.
+
+## hubconf.py
+`hubconf.py` should contain a `class Model` with the following structure:
+
+```
+class Model:
+    device = None
+    jit = False
+
+    def set_device(self, device):
+        self.device = device
+
+    def set_jit(self, jit: bool):
+        self.jit = jit
+
+    def get_module(self):
+        """ Returns `model`, `example_input`
+        `model` should be torchscript model if self.jit is True
+        Both `model` and `example_input` should be on `self.device` properly."""
+
+    def get_eval_function(self):
+        """ Returns function `f` that triggers model evaluation that finishes in 30s.
+        `f` can be called with f(model, example_input)"""
+
+    def get_training_loop(self):
+        """ Returns a function `f` that triggers model training that finishes in 30s.
+        `f` can be called with f(model, example_input)"""
+```
+To test locally before submitting a PR, add the following snippet to `hubconf.py` and run `python hubconf.py`.
+```
+if __name__ == '__main__':
+    m = Model()
+    m.set_device('cpu')
+    model, example_input = m.get_module()
+    m.get_training_loop()(model, example_input)
+    m.get_eval_function()(model, example_input)
```

Comment:
Similar to above, `model` is not going to be clear for some examples since there might be multiple `nn.Module` objects involved. Though for eval having an example input may be useful.